src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -197,26 +197,17 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
     return filters.reduce((items, filter) => filter(items), items);
   }
 
-  @computed get allItems() {
-    const { filterItems, store } = this.props;
-
-    return this.applyFilters(filterItems, store.items);
-  }
-
   @computed get items() {
-    const { allItems, filters, filterCallbacks } = this;
-    const filterItems: ItemsFilter[] = [];
+    const {filters, filterCallbacks } = this;
     const filterGroups = groupBy<Filter>(filters, ({ type }) => type);
 
-    Object.entries(filterGroups).forEach(([type, filtersGroup]) => {
-      const filterCallback = filterCallbacks[type];
-
-      if (filterCallback && filtersGroup.length > 0) {
-        filterItems.push(filterCallback);
-      }
-    });
+    const filterItems = Object.keys(filterGroups)
+      .map(type => [filterCallbacks[type], filterGroups[type]] as const)
+      .filter(([callback, groups]) => callback && groups.length)
+      .map(([callback]) => callback)
+      .concat(this.props.filterItems);
```

Comment:
This refactoring with multiple iterations doesn't give code cleanliness, it's harder to understand what is intent here. Also, `as const` part looks like a forced measure, not the planned thing.

@pashevskii @jim-docker @Nokel81 Maybe reconsider once more if we should choose #2114 (or refactror this part once more)?